api/v1alpha2/loadbalancer_types.go

```diff
@@ -0,0 +1,132 @@
+}
+
+// +kubebuilder:object:root=true
+
+// LoadBalancerList contains a list of Loadbalancer
+type LoadBalancerList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []LoadBalancer `json:"items"`
+}
+
+// LoadBalancerSpec defines the desired state of the load balancer
+type LoadBalancerSpec struct {
+
+	// Ports ports that should be used in this load balancer
+	Ports []LoadBalancerPort `json:"ports"`
+
+	// Selector selects the machines to add to the load balancing pool
+	Selector map[string]string `json:"selector"`
+
+	// ConfigRef holds a reference to a provider-specific configuration
+	ConfigRef corev1.ObjectReference `json:"configRef"`
+}
+
+// LoadBalancerPort describes the desired state of load balancer ports
+type LoadBalancerPort struct {
+
+	// Name is the name of this port. This is used by the provider to guarantee uniquness
+	// of the <Protocol, Port, TargetPort> association
+	Name string `json:"name"`
+
+	// The IP protocol for this port. Supports "TCP" for now.
+	// Default is TCP.
+	// +optional
+	Protocol *Protocol `json:"protocol,omitempty"`
+
+	// The port that will be exposed by this load balancer.
+	Port int32 `json:"port"`
+
+	// TargetPort is the port of the machines the load balancer should send traffic to
+	// +optional
+	TargetPort *int32 `json:"targetPort,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=awsloadbalancerconfigs,scope=Namespaced,categories=cluster-api
+// +kubebuilder:storageversion
+
+// AWSLoadBalancerConfig is the schema for the aws configuration API
+type AWSLoadBalancerConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec AWSLoadBalancerConfigSpec `json:"spec,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// AWSLoadBalancerConfigList contains a list of AWSLoadBalancerConfig
+type AWSLoadBalancerConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []AWSLoadBalancerConfig `json:"items"`
+}
+
+// AWSLoadBalancerConfigSpec defines the desired state of an AWS load balancer
+type AWSLoadBalancerConfigSpec struct {
+	// VpcID is the id of the VPC used to create loadBalancers
+	VpcID string `json:"vpcId"`
+
+	// SubnetIDs is the list of subnets within the VPC used
+	SubnetIDs []string `json:"subnetIds"`
+
+	// Region is the region used for the loadbalancers
+	Region string `json:"region"`
+}
+
+// Protocol defines listener protocols for a classic load balancer.
+type Protocol string
+
+var (
+	// ProtocolTCP defines the API string representing the TCP protocol
+	ProtocolTCP = Protocol("TCP")
+)
+
+// LoadBalancerStatus defines the observed state of the AWS load balancer
+type LoadBalancerStatus struct {
+
+	// APIEndpoint represents the endpoint to communicate with the load
+	// balancer.
+	// +optional
+	APIEndpoint APIEndpoint `json:"apiEndpoint,omitempty"`
```

Comment:
Not doing a full review on this right now (unless you'd like), but one thing I wanted to mention was the API endpoint needs to be in spec, not status. If you have a provider that generates an endpoint (e.g. a DNS name) **and** the endpoint is not something you can determine by querying, we would lose the data if we store it in status and we're trying to move the resource from one management cluster to another.

Reply:
@ncdc - I'll rebase and ping you for an API review